packages/api/src/task/scheduler.ts

```diff
@@ -409,20 +433,23 @@ export class TaskScheduler {
     if (typeof asset === "string") {
       asset = await db.asset.get(asset);
     }
-    await db.asset.markDeleted(asset.id);
-    await this.queue.publishWebhook("events.asset.deleted", {
-      type: "webhook_event",
-      id: uuid(),
-      timestamp: Date.now(),
-      event: "asset.deleted",
-      userId: asset.userId,
-      payload: {
-        asset: {
+
```

Comment:
You should probably check if the asset doesn't already have `phase: deleted`, in which case we should either return an error like a 404. Better yet, this function is only ever called with a full `Asset`, never with a `string`. We should remove the `string` parameter here and make all the checks we need on the controller side, together with the place we check the asset belongs to the user that is actually calling a delete on it. There we should check:
 - that the asset is not already `phase: deleted`, in which case we should return a 404
 - otherwise, if the asset has `deleted: true`, only allow "redeleting" if user is an admin (so we can use this to cleanup old assets)